test/config.test.js

```diff
@@ -1194,140 +1194,6 @@ usePathAsTransactionNameTests.forEach(function (usePathAsTransactionNameTest) {
   );
 });
 
-test('disableInstrumentations', function (t) {
```

Comment:
Reviewer note: I have dropped this test. It has always been a thorn. All this knowledge in here for dropping certain packages (because of node version compat) from the test is silly. The test provides basically nil value. It also requires deep internal knowledge of "lib/instrumentation/index.js" impl details.

Reply:
I agree is cumbersome to keep this test up to date and thanks to this PR I think it should be easier since instrumentation enablement relies on the proper grouping of modules and it sub-modules by their `diKey`.

Perhaps a test on `PatchRegistry` ensuring keys are properly resolved would be a good substitute for this test.

Reply:
I added some testing in commit [eeed31c](https://github.com/elastic/apm-agent-nodejs/pull/3658/commits/eeed31c95337c30e25243e81eff2e2a88a2a5acd)